torchqdynamics/sesolve/euler.py

```diff
@@ -0,0 +1,25 @@
+import torch
+
+from ..odeint import ForwardQSolver
+from ..solver import SolverOption
+from ..types import TimeDependentOperator
+
+
+class SEEuler(ForwardQSolver):
```

Comment:
Actually it's maybe an `AdjointQSolver`, but let's not care about implementing the adjoint on this guy for now, except if you find it useful for some reason @gautierronan (maybe as a playground example?).

Reply:
So I did not think about it too much tbh. It can definitly be implemented for `sesolve`, the question is more: is it useful?

The adjoint method is an answer to the large memory requirement of regular backward autodiff for `mesolve`. For `sesolve`, ket takes much less memory, but the Hamiltonian is also stored in the graph of operations I believe. In this case, one could still run quite quickly in memory problems for time-dependent problems. So I guess it could also be useful, and it's quite easy to implement.

Reply:
I agree, at least it's a good and simple example to explain how the adjoint works.